src/Components/CreateDogList/CreateDogList.js

```diff
@@ -1,90 +1,55 @@
-import React, { useEffect, useState } from 'react';
-import { sample } from 'lodash';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import './CreateDogList.styles.scss';
 import { connect, useDispatch } from 'react-redux';
 import InfiniteScroll from 'react-infinite-scroll-component';
 import Loading from '../Loading/Loading';
 import DogCardTemplateList from '../DogCardTemplateList/DogCardTemplateList';
-import { setLoading } from '../../Redux/Loading/LoadingSlice';
-import { setDataDog } from '../../Redux/DogResults/DogResultsRedux';
-import apiDogCall from '../../api/dogs/ApiDogCall';
-import UrlParameters from '../../api/dogs/UrlParameters';
-import apiNameCall from '../../api/name/apiNameCall';
+import { fetchNameDataStart } from '../../Redux/Names/NamesRedux';
+import { getDogData } from '../../const/selectors/selectorDogResults';
+import { changeLoading, updateOffset } from '../../Redux/Dog/DogResultsRedux';
 
-function CreateDogList({ isLoading }) {
+function CreateDogList({ isDogLoading }) {
   const dispatch = useDispatch();
-  const setLoadingTrue = () => {
-    dispatch(setLoading(true));
-  };
-  const [dogNames, setDogNames] = useState([]);
-  const [dogInfos, setDogInfos] = useState([]);
-  const [combinedDogData, setCombinedDogData] = useState([]);
   const [offsetAmount, setOffsetAmount] = useState(0);
   const dataContinues = true;
-  const parameters = UrlParameters();
   const limitperPage = 20;
-
-  //* After both Requests have been completed, we merge them both into a
-  //* singular array
-  const mergeApiDatas = (names, infos) => {
-    const combinedList = infos.map((dg) => {
-      const Names = sample(names);
-      return { ...dg, nameDog: Names };
-    });
-    setCombinedDogData(combinedList);
-    dispatch(setDataDog(combinedList));
-    return combinedList;
-  };
-
-  //* Requests a random name, its currently set for 50 names
-  const randomNameInfo = async () => {
-    await apiNameCall().then((results) => setDogNames(results));
+  const completedDogData = getDogData();
+
+  //* This function will execute when button is pressed
+  //* it will activate the saga cycle
+  const fetchDogApiData = async () => {
+    dispatch(changeLoading(true));
```

Comment:
loading should be inside of the saga, because here you are not controlling side effects
https://redux.js.org/tutorials/fundamentals/part-6-async-logic#redux-middleware-and-side-effects
```
Redux Middleware and Side Effects
By itself, a Redux store doesn't know anything about async logic. It only knows how to synchronously dispatch actions, update the state by calling the root reducer function, and notify the UI that something has changed. Any asynchronicity has to happen outside the store.

Earlier, we said that Redux reducers must never contain "side effects". A "side effect" is any change to state or behavior that can be seen outside of returning a value from a function. Some common kinds of side effects are things like:

Logging a value to the console
Saving a file
Setting an async timer
Making an AJAX HTTP request
Modifying some state that exists outside of a function, or mutating arguments to a function
Generating random numbers or unique random IDs (such as Math.random() or Date.now())
```